PYME/Acquire/Hardware/Simulator/wormlike2.py

```diff
@@ -42,45 +41,45 @@ class wormlikeChain:
     def __init__(self, kbp, steplength=10.0, lengthPerKbp=10.0, persistLength=150.0):
         numsteps = int(round(lengthPerKbp*kbp/steplength))
 
-        exp_costheta = (exp(-steplength/persistLength));
-        theta = sqrt(2*log(1/exp_costheta))*abs(randn(numsteps));
-        phi = 2*pi*rand(numsteps);
+        exp_costheta = (np.exp(-steplength/persistLength));
+        theta = np.sqrt(2*np.log(1/exp_costheta))*abs(np.random.randn(numsteps));
+        phi = 2*np.pi*np.random.rand(numsteps);
         #phi = 0.5*pi*randn(numsteps, 1)+pi/20;
 
-        phi = cumsum(concatenate(([0], phi),0))
+        phi = np.cumsum(np.concatenate(([0], phi),0))
         
-        xs = 1 - 2*rand()
-        ys = 1 - 2*rand()
-        zs = 1 - 2*rand()
+        xs = 1 - 2*np.random.rand()
+        ys = 1 - 2*np.random.rand()
+        zs = 1 - 2*np.random.rand()
 
-        nrm = sqrt(xs**2 + ys**2 + zs**2)
+        nrm = np.sqrt(xs**2 + ys**2 + zs**2)
 
         xs = xs/nrm;
         ys = ys/nrm;
         zs = zs/nrm;
 
-        so = array([xs, ys, zs])
+        so = np.array([xs, ys, zs])
 
-        xs = steplength*xs*ones(theta.shape)
-        ys = steplength*ys*ones(theta.shape)
-        zs = steplength*zs*ones(theta.shape)
+        xs = steplength*xs*np.ones(theta.shape)
+        ys = steplength*ys*np.ones(theta.shape)
+        zs = steplength*zs*np.ones(theta.shape)
 
         for i in range(2,numsteps):
-            sh = cross(so, so + array([0,0,2]))
+            sh = np.cross(so, so + np.array([0,0,2]))
             #sh = sh./sqrt(dot(sh, sh));
             #sh = sh./sqrt(sh*sh.');
-            sh = sh/dot(sh, sh.T)
-            sk = cross(so, sh)
+            sh = sh/np.dot(sh, sh.T)

```

Comment:
I recall there being a difference between the `scipy.dot`, `numpy.dot`, and `numpy.linalg.dot` (likewise with `cross`). Are these really a 1:1 replacement and have we checked that this still works (don't think it's covered by any of the unit tests).

Reply:
`scipy.dot` is pretty tough to find documentation for... But opening in ipython:
```
Help on function dot in module numpy:

dot(...)
    scipy.dot is deprecated and will be removed in SciPy 2.0.0, use numpy.dot instead
```

and cross
```
In [4]: help(scipy.cross)                                                                                                                    

Help on function cross in module numpy:

cross(a, b, axisa=-1, axisb=-1, axisc=-1, axis=None)
    scipy.cross is deprecated and will be removed in SciPy 2.0.0, use numpy.cross instead
```
which would at least has the same signature as the numpy version:
```
Help on function cross in module numpy:

cross(a, b, axisa=-1, axisb=-1, axisc=-1, axis=None)
    Return the cross product of two (arrays of) vectors.
    
    .....
```

Note also that this particular wormlike, `wormlike2`, gets used in `tests/PYME/localization/test_MultifitSR.py`, so we do actually have coverage for it, kind of.